### PR TITLE
Prune access from access-inactive users

### LINKED ISSUE #3695: Revoke elevated access from access-inactive users
Daily job to remove any attached roles and permissions from users that are not access active

Non-member role is already reattached automatically if needed 

Also remove PM relationship from teams

Sync access to Jedi when done

### PR DESCRIPTION
**Pull request by kberzinch**
Closes #3695

### CODE REVIEW

File: app/Jobs/PruneAccessFromAccessInactiveUsers.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                $user->roles()->detach();
+                $user->permissions()->detach();
+
+                $user->manages()
+                    ->update(['project_manager_id' => null]);
+
+                $user->novaNotifications()
+                    ->delete();
+
+                $user->searchable();
```
Comment (JoshOldenburg) [inline app/Jobs/PruneAccessFromAccessInactiveUsers.php:52]: What does this do?

Comment (kberzinch) [reply app/Jobs/PruneAccessFromAccessInactiveUsers.php:52]: Reindexes the user in Meilisearch (since some places use role filters in Meilisearch)

**Review by evan10s — COMMENTED** [review]
Given that if access ends unintentionally early, it would create a decent amount of human effort to restore all PMs/trainer/team/admin access in apiary, we should probably add a grace period in this job.

JEDI doesn't wait, but its restoration process is automated. I think we should add at least a 24 hour grace period before removing anything from Apiary.

Also, can we disable this behavior on a config flag for non-prod environments? It would be annoying to have to keep up access active statuses or manually reset access in tinker for that.

**Comment by kberzinch** [comment]
I'll add a feature flag. Are you OK with merging this in combination with #4408?

**Comment by evan10s** [comment]
> I'll add a feature flag. Are you OK with merging this in combination with #4408?

Even with the notification beforehand, I think the grace period is still prudent

**Review by evan10s — APPROVED** [review]

**Review by JoshOldenburg — COMMENTED** [review]
I don't see where you're actually using the new timestamp logic. All I see is the `! $user->is_access_active` in the job. Am I missing something?

File: app/Jobs/PruneAccessFromAccessInactiveUsers.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        $this->queue = 'jedi';
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle(): void
+    {
+        User::accessInactive(CarbonImmutable::now()->subDay())
```
Comment (kberzinch) [inline app/Jobs/PruneAccessFromAccessInactiveUsers.php:36]: The timestamp passed in here is 1 day ago, so the DB results will contain users that were inactive as of 1 day ago

File: app/Jobs/PruneAccessFromAccessInactiveUsers.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            ->where('is_service_account', '=', false)
+            ->where(static function (Builder $query): void {
+                $query->whereHas('roles')
+                    ->orWhereHas('permissions');
+            })
+            ->get()
+            ->filter(static fn (User $user): bool => ! $user->is_access_active)
```
Comment (kberzinch) [inline app/Jobs/PruneAccessFromAccessInactiveUsers.php:43]: The check down here is to make sure the users are _still_ access-inactive. I can't easily chain the same scope on itself so this seemed like the least bad option.

**Review by kberzinch — COMMENTED** [review]
See below inline comments

File: app/Jobs/PruneAccessFromAccessInactiveUsers.php (hunk 2 above)
Comment (JoshOldenburg) [reply app/Jobs/PruneAccessFromAccessInactiveUsers.php:36]: This is the line I was missing. Thanks.

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**